src/disco/components/App/index.js

```diff
@@ -36,21 +47,57 @@ export class AppBase extends React.Component {
           <Helmet defaultTitle={i18n.gettext('Discover Add-ons')}>
             <meta name="robots" content="noindex" />
           </Helmet>
-          <ErrorPage>{children}</ErrorPage>
+
+          <ErrorPage>
+            <Switch>
+              <Route exact path={DISCO_PANE_PATH} component={DiscoPane} />
+              <Route exact path="/:lang/firefox/404" component={NotFound} />
+              <Route
+                exact
+                path="/:lang/firefox/500"
+                component={
+                  config.get('isDevelopment') ? GenericError : NotFound
+                }
+              />
+              <Route
+                exact
+                path="/:lang/:app/simulate-async-error/"
+                component={SimulateAsyncError}
+              />
+              <Route
+                exact
+                path="/:lang/:app/simulate-client-error/"
+                component={SimulateClientError}
+              />
+              <Route
+                exact
+                path="/:lang/:app/simulate-sync-error/"
+                component={SimulateSyncError}
+              />
+              <Route component={NotFound} />
+            </Switch>
+          </ErrorPage>
+
           <Footer />
         </div>
       </NestedStatus>
     );
   }
 }
 
-export function mapStateToProps(state, ownProps) {
+function mapStateToProps(state, ownProps) {
+  const match = matchPath(ownProps.location.pathname, {
+    path: DISCO_PANE_PATH,
+    exact: true,
+  });
+
   return {
-    browserVersion: ownProps.params.version,
+    browserVersion: match.params.version,
```

Comment:
Can't this be `ownProps.match.params.version`?

Reply:
No because it is outside a `Route`, that is why I have to use the `matchPath()` function.